.github/workflows/build.yaml

```
@@ -51,12 +51,47 @@ jobs:
     - name: Run Tests
       run: |
         make test
-  docker-build:
-    name: docker-build
+
+  docker-build-amd64:
+    name: docker-build-amd64
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            dataplane:
+              - 'dataplane/**'
+            containerfile:
+              - 'build/Containerfile.dataplane'
+      - name: Setup QEMU
+        if: steps.filter.outputs.dataplane || steps.filter.outputs.containerfile
+        uses: docker/setup-qemu-action@v3
+      - name: Setup Docker Buildx
+        if: steps.filter.outputs.dataplane || steps.filter.outputs.containerfile
+        id: buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          buildkitd-flags: "--debug"
+      - name: Build container image for amd64
+        if: steps.filter.outputs.dataplane || steps.filter.outputs.containerfile
+        uses: docker/build-push-action@v6
+        with:
+          push: false
+          builder: ${{ steps.buildx.outputs.name }}
+          context: .
+          platforms: linux/amd64
+          file: build/Containerfile.dataplane
+          tags: kong/blixt-dataplane:pr-${{ github.event.pull_request.number }}-amd64-${{ github.sha }}
+
+  docker-build-arm64:
+    name: docker-build-arm64
     runs-on: ubuntu-22.04
```

```suggestion
    runs-on: ubuntu-22.04-arm
```

this needs to run on an arm runner since we want to build for the native architecture of the machine (arm64 in this case).
amazing timing for this PR, github launched free arm64 runners for oss repos two days ago :))
https://github.blog/changelog/2025-01-16-linux-arm64-hosted-runners-now-available-for-free-in-public-repositories-public-preview/

---

ohh thats nice,i will change it

---

and also we dont need qemu anynore then?

---

Damnn,it took 5 min now 😆